tests/RobotFramework/tests/mqtt/disconnect_from_broker.robot

```diff
@@ -0,0 +1,71 @@
+*** Settings ***
+Resource        ../../resources/common.resource
+Library         ThinEdgeIO
+Library         OperatingSystem
+
+Suite Setup     Custom Setup
+Suite Teardown  Get Logs
+
+Test Tags      theme:mqtt
+
+
+*** Variables ***
+
+${TOPIC}             test/topic
+${MESSAGE}           Hello
+
+
+*** Test Cases ***
+
+Validate no Error message in journalctl
+    [Documentation]    This Test step is inspired by 'tedge mqtt cli commands do not disconnect cleanly from MQTT broker #2922' bug
+    ...                extended to check for any possible error that may appear in the journalctl
+    Set up broker with server and client authentication
+    Restart Service    mosquitto
+    tedge configure MQTT server authentication
+    tedge configure MQTT client authentication
+    Execute Command    tedge mqtt pub foo bar
+    ${log}    Execute Command    journalctl -u mosquitto -n 30
+    Should Not Contain    ${log}    Error
+
+Publish Connection Closed On Interruption
+    [Documentation]    Tests that a publish connection to MQTT broker closes upon interruption.
+    Set up broker with server and client authentication
+    Restart Service    mosquitto
+    tedge configure MQTT server authentication
+    tedge configure MQTT client authentication
+    ${pub_process}=    Execute Command    tedge mqtt pub ${TOPIC} ${MESSAGE} & echo $!    return_stdout=True
+    Sleep    2s    reason=Allow some time for connection
+    Kill Process    ${pub_process}
+    Process Should Not Be Running    ${pub_process}
+    Log    MQTT publish connection was closed upon interruption.
+
+Subscribe Connection Closed On Interruption
+    [Documentation]    Tests that a subscribe connection to MQTT broker closes upon interruption.
+    ${sub_process}=    Execute Command    tedge mqtt sub ${TOPIC} & echo $!   return_stdout=True
+    Sleep    2s    reason=Allow some time for connection
+    Kill Process    ${sub_process}
+    Process Should Not Be Running    ${sub_process}
+    Log    MQTT subscribe connection was closed upon interruption.
+
+
+*** Keywords ***
+Custom Setup
+    Setup    skip_bootstrap=True
+    Execute Command    /setup/bootstrap.sh --no-connect --no-secure
+
+Set up broker with server and client authentication
+    Transfer To Device    ${CURDIR}/mosquitto-client-auth.conf    /etc/tedge/mosquitto-conf/mosquitto-client-auth.conf
+    Execute Command
+    ...    mv /etc/tedge/mosquitto-conf/mosquitto-client-auth.conf /etc/tedge/mosquitto-conf/tedge-mosquitto.conf
```

Comment:
Why not directly move the file to its destination ?

Reply:
Is this file actually needed? The bootstrapping of the test should take care of the mosquitto setup, so I don't think we need the extract mosquitto-client-auth.conf setup. Port `1883` should be fine for these tests